src/main/java/io/cdap/plugin/mainframe/config/MainframeRecordReaderConfig.java

```
@@ -0,0 +1,541 @@
+    .put("250K", 250000)
+    .put("500K", 500000)
+    .put("750K", 750000)
+    .put("1M", 1000000)
+    .put("2M", 2000000)
+    .build();
+
+  public MainframeRecordReaderConfig(String referenceName, String copybook, String inputPath, String charset,
+                                     String fileOrganization, String dialect, String copybookSplit,
+                                     @Nullable String recordDeciderFieldName,
+                                     @Nullable String recordDeciderConditions,
+                                     @Nullable String recordSelectorFieldMappings,
+                                     @Nullable String recordSelectorExpression,
+                                     @Nullable Boolean recordSelectorUnmatchedToError,
+                                     @Nullable String recordsPerPartition) {
+    super(referenceName);
+    this.copybook = copybook;
+    this.charset = charset;
+    this.inputPath = inputPath;
+    this.fileOrganization = fileOrganization;
+    this.dialect = dialect;
+    this.copybookSplit = copybookSplit;
+    this.recordDeciderFieldName = recordDeciderFieldName;
+    this.recordDeciderConditions = recordDeciderConditions;
+    this.recordSelectorFieldMappings = recordSelectorFieldMappings;
+    this.recordSelectorExpression = recordSelectorExpression;
+    this.recordSelectorUnmatchedToError = recordSelectorUnmatchedToError;
+    this.recordsPerPartition = recordsPerPartition;
+  }
+
+  /**
+   * @return a string representation of copybook.
+   */
+  public String getCopybook() {
+    return copybook;
+  }
+
+  /**
+   * @return true if copybook is a macro, false otherwise.
+   */
+  public boolean isCopybookMacro() {
+    return containsMacro(PROPERTY_COPYBOOK);
+  }
+
+  /**
+   * @return Encoding to be used for the parsing the binary file.
+   */
+  public String getCharset() {
+    return charset == null || !charsetMappings.containsKey(charset) ? "cp037" : charsetMappings.get(charset);
```

if the user specified charset doesn't exist, shouldn't that be a validation error instead of silently using something else?

---

Is not nullable. I will remove this check.